config/peerpods/podvm/libvirt-podvm-image-handler.sh

```diff
@@ -0,0 +1,301 @@
+        error_exit "Failed to add the libvirt uuid to peer-pods-cm configmap"
+    echo "Libvirt pool id added as annotation to peer-pods-cm configmap successfully"
+
+}
+
+# function to delete the libvirt
+# libvirt_id must be set as an environment variable
+
+function delete_libvirt_image_using_uuid() {
+    echo "Deleting Libvirt image"
+
+    # Delete the Libvirt pool 
+    # If any error occurs, exit the script with an error message
+
+    # LIBVIRT_POOL_UUID shouldn't be empty
+    [[ -z "${LIBVIRT_POOL_UUID}" ]] && error_exit "LIBVIRT_POOL_UUID is empty"
+
+    # Delete the libvirt pool
+    echo "Deleting libvirt pool."
+    virsh -d 0 -c $LIBVIRT_URI pool-destroy $LIBVIRT_POOL_UUID --sparse ||
+        error_exit "Failed to destroy the libvirt pool"
+    
+    virsh -d 0 -c $LIBVIRT_URI pool-undefine $LIBVIRT_POOL_UUID --sparse ||
+        error_exit "Failed to undefine the libvirt pool"
+
+    echo "Uploaded the image successfully"
+
+    # Remove the ami id annotation from peer-pods-cm configmap
+    delete_libvirt_uuid_annotation_from_peer_pods_cm
+
+}
+
+# Function to delete the LIBVIRT_POOL_UUID annotation from the peer-pods-cm configmap
+
+function delete_libvirt_uuid_annotation_from_peer_pods_cm() {
+    echo "Deleting ami id annotation from peer-pods-cm configmap"
+
+    # Check if the peer-pods-cm configmap exists
+    if ! kubectl get configmap peer-pods-cm -n openshift-sandboxed-containers-operator >/dev/null 2>&1; then
+        echo "peer-pods-cm configmap does not exist. Skipping deleting the ami id"
+        return
+    fi
+
+    # Delete the libvirt uuid annotation from peer-pods-cm configmap
+    kubectl annotate configmap peer-pods-cm -n openshift-sandboxed-containers-operator \
+        "LIBVIRT_POOL_UUID" ||
+        error_exit "Failed to delete the ami id annotation from peer-pods-cm configmap"
+
+    echo "libvirt uuid annotation deleted from peer-pods-cm configmap successfully"
+}
+
+# display help message
+
+function display_help() {
+    echo "This script is used to create libvirt qcow2 for podvm"
+    echo "Usage: $0 [-c|-C] [-- install_binaries]"
+    echo "Options:"
+    echo "-c  Create image"
+    echo "-C  Delete image"
+}
+
+function install_packages(){
+    if [[ -n "${ACTIVATION_KEY}" && -n "${ORG_ID}" ]]; then
+        subscription-manager register --org=${ORG_ID} --activationkey=${ACTIVATION_KEY}; \
+    fi
+
+    subscription-manager repos --enable codeready-builder-for-rhel-9-${ARCH}-rpms; \
+    dnf install -y libvirt-client; \
+    dnf groupinstall -y 'Development Tools'; \
+    dnf install -y jq; \
+    dnf install -y yum-utils gnupg git --allowerasing curl pkg-config clang perl libseccomp-devel gpgme-devel \
+    device-mapper-devel qemu-kvm unzip wget libassuan-devel genisoimage cloud-utils-growpart cloud-init;
+
+    curl -L -o /usr/local/bin/yq https://github.com/mikefarah/yq/releases/download/${YQ_VERSION}/yq_linux_${ARCH/x86_64/amd64} \
```

Comment:
I would recommend updating the install_binary_packages() in lib.sh to support multi-arch.
For binaries which you need to build may be have a separate method in lib.sh and use it in the handler.

Reply:
If some binaries can be built downstream and made part of the builder image it will be good. Otherwise the process will be so time consuming. Wdyt @snir911 ?

Reply:
I agree, however, @supritai said binaries are taken from payload image, hence, I'm not sure what is left to be built here, IIUC most of the preparations in this file can be dropped (these are taken from the upstream Dockerfile.podvm_builder*)

Reply:
Included only needed packages.